esp/src/src-react/components/Permissions.tsx

```diff
@@ -0,0 +1,210 @@
+
+const defaultUIState = {
+    canDelete: false,
+    categorySelected: false,
+    fileScope: false,
+    hasSelection: false,
+    repositoryModule: false,
+    workunitScope: false
+};
+
+interface PermissionsProps {
+}
+
+export const Permissions: React.FunctionComponent<PermissionsProps> = ({
+}) => {
+
+    const [grid, setGrid] = React.useState<any>(undefined);
+    const [selection, setSelection] = React.useState([]);
+    const [showAddPermission, setShowAddPermission] = React.useState(false);
+    const [scopeScansEnabled, setScopeScansEnabled] = React.useState(false);
+    const [modulesDn, setModulesDn] = React.useState("");
+    const [uiState, setUIState] = React.useState({ ...defaultUIState });
+
+    //  Grid ---
+    const gridStore = useConst(WsAccess.CreatePermissionsStore(null, null));
+    const gridSort = useConst([{ attribute: "name", "descending": false }]);
+    const gridQuery = useConst({});
+    const gridColumns = useConst({
+        check: selector({
+            width: 27,
+            disabled: function (row) {
+                if (row.name === "File Scopes" || row.name === "Workunit Scopes" || row.name === "Repository Modules") {
+                    return false;
+                }
+                return row.children ? true : false;
+            }
+        }, "checkbox"),
+        name: tree({
+            width: 360,
+            label: nlsHPCC.Name,
+            formatter: function (_name, idx) {
+                if (idx.__hpcc_parent) {
+                    return `<a href="/#/security/permissions/${_name}/${idx.__hpcc_parent.name}">${_name}</a>`;
+                } else {
+                    return _name;
+                }
+            }
+        }),
+        description: { label: nlsHPCC.Description },
+        basedn: { label: "basedn" }
+    });
+
+    //  Selection  ---
+    React.useEffect(() => {
+        const state = { ...defaultUIState };
+        if (selection.length) {
+            state.hasSelection = true;
+            if (!selection.filter(item => item.hasOwnProperty("children")).length) {
+                state.canDelete = true;
+            } else {
+                state.categorySelected = true;
+            }
+            if (selection.filter(item => item.name === "File Scopes").length) {
+                const fileScope = selection.filter(item => item.name === "File Scopes")[0];
+                state.fileScope = true;
+                if (fileScope) {
+                    WsAccess.Resources({
+                        request: {
+                            name: fileScope?.data?.Basedn
+                        }
+                    })
+                        .then(({ ResourcesResponse }) => {
+                            setScopeScansEnabled(ResourcesResponse?.scopeScansStatus?.isEnabled || false);
+                        })
+                        .catch(logger.error)
+                        ;
+                }
+            }
```

Comment:
I think I would move this entire `if` block into its own React useEffect function - as its a bit unusual to have an Async call inside the selection code?